src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/controller/TriggerProcessingCrnController.kt

```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.digital.hmpps.hmppsallocations.controller
+
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.reactive.asFlow
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.http.codec.multipart.FilePart
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestPart
+import org.springframework.web.bind.annotation.RestController
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.hmppsallocations.service.TriggerReprocessService
+
+@RestController
+@RequestMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
+class TriggerProcessingCrnController(private val triggerReprocessService: TriggerReprocessService) {
+
+  @PostMapping("/crn/upload")
+  suspend fun uploadCrns(@RequestPart("file") file: Mono<FilePart>): ResponseEntity<Void> {
+    triggerReprocessService.sendEvents(fileToCases(file))
+    return ResponseEntity.ok().build()
+  }
+
+  suspend fun fileToCases(filePart: Mono<FilePart>): List<String> {
+    return filePart.flatMapMany { file ->
+      file.content().flatMapIterable { dataBuffer ->
+        dataBuffer.asInputStream().bufferedReader().use { reader ->
+          reader.lineSequence()
+            .filter { it.isNotBlank() }
+            .toList()
+        }
+      }
+    }.asFlow().toList()
```

Comment:
Is `asFlow` needed - probably, just not sure if Flow needs to be introduced here - if we have to use `Mono and we are converting to a list anyway.

Reply:
removing this and changing it `.toIterable()` (there is no Flux to List method) causes `Exception in thread "DefaultDispatcher-worker-1 @coroutine#2" org.springframework.core.codec.DecodingException: Could not find first boundary` to happen, I'm guessing its to do with when this is executed (this would be executed before the response is returned, where as the other way would be executed after the response is returned?